### DISCUSSION TITLE
custom link modal

### DISCUSSION THREAD

**Discussion by SunnyCheung-cp** [Milkdown/milkdown#1105] [Q&A] [answered]
I want to implement a function,  when I type in "testing" and I select it then click the link button, will show the pop up
But i don't know how to get the selected text and update it

<img width="559" alt="image" src="https://github.com/Milkdown/milkdown/assets/105030733/4be85826-a557-4c14-9317-aecfa1aef0a8">

**Comment by Saul-Mirone** [comment]
You can always get selection from `editorState.selection`, you'll get the position of your selection and you can get nodes from it.

**Comment by sytolk** [comment]
@SunnyCheung-cp you can check this Dialog: https://github.com/tagspaces/tagspaces-common/blob/274a194a653154ff2249ae1e65a05ea3de2450a8/packages/tagspaces-md/src/components/dialogs/LinkDialog.tsx#L25

here is selected text for Dialog: https://github.com/tagspaces/tagspaces-common/blob/d8a4daf098d2fef9e4a6af997fe712540fb1818b/packages/tagspaces-md/src/components/LinkTooltip/LinkTooltip.tsx#L55

https://github.com/tagspaces/tagspaces-common/blob/a0660593f7fb3fe193f52fce545b1c1dbf307588/packages/tagspaces-md/src/hooks/useSelectedMarkPosition.tsx#L22

**Comment by zgrybus** [comment, marked as answer]
@SunnyCheung-cp 
Feel free to use https://github.com/DareData/notion-style-editor

**Reply by SunnyCheung-cp** [reply]
@zgrybus  Awesome, Thanks for your help.
I tried the demo and if I use the link in the first row it stays there until I remove it starting from the second row


https://github.com/Milkdown/milkdown/assets/105030733/59a36451-a67b-4db5-b641-3a493990af29
